### PR TITLE
Add support for dunai 0.11

### PR DESCRIPTION
**Pull request by blackheaven**
I'm a bit struggling with instances:

```
src/FRP/Rhine/ClSF/Util.hs:338:19: error:
    • Could not deduce (Floating
                          (MSF (Control.Monad.Trans.Reader.ReaderT (TimeInfo cl) m) v s))
        arising from a use of ‘^-^’
      from the context: (Monad m, Eq s, Floating s, VectorSpace v s,
                         s ~ Diff td, td ~ Time cl)
        bound by the type signature for:
                   highPass :: forall (m :: Type -> Type) s v td.
                               (Monad m, Eq s, Floating s, VectorSpace v s, s ~ Diff td) =>
                               Diff td -> BehaviourF m td v v
        at src/FRP/Rhine/ClSF/Util.hs:(328,1)-(337,21)
    • In the expression: clId ^-^ lowPass t
      In an equation for ‘highPass’: highPass t = clId ^-^ lowPass t
    |
338 | highPass t = clId ^-^ lowPass t
    |                   ^^^
```

which is weird since:

```
instance (Monad m, Eq s, Num s, VectorSpace v s, Floating (MSF m a s)) => VectorSpace (MSF m a v) (MSF m a s)
instance (Monad m, Floating b) => Floating (MSF m a b)
```

If anyone can have a look

### CODE REVIEW

File: rhine-bayes/rhine-bayes.cabal [hunk 1]
```diff
@@ -30,10 +30,10 @@ library
     FRP.Rhine.Bayes
   other-modules:
     Data.MonadicStreamFunction.Bayes
-  build-depends:       base         >= 4.11 && < 4.18
```
Comment (turion) [inline rhine-bayes/rhine-bayes.cabal:33]: I'd rather keep tight version bounds on `base`, so one should bump to 4.19 here.

**Comment by turion** [comment]
> I'm a bit struggling with instances:

Ah yes, that one is really tricky, it took me some time. You need:
```
import Data.MonadicStreamFunction.Instances.Num ()
```

**Comment by turion** [comment]
I think you're actually bumping the dunai version, and not so much the GHC version, right?

**Comment by blackheaven** [comment]
> Ah yes, that one is really tricky, it took me some time. You need:

Thank you so much!

**Comment by blackheaven** [comment]
> I think you're actually bumping the dunai version, and not so much the GHC version, right?

and `simple-affine-space`, but that's the only blockers to GHC 9.4 support.

**Comment by turion** [comment]
But GHC 9.4 is already supported in `rhine`. I think to support _9.6_, one needs to bump `dunai`, but I'm not sure. The issue with 9.4 is only that the dependency `monad-bayes` didn't support 9.4 for a long time. But it does now, I think we just need to enable the CI jobs again.

**Comment by blackheaven** [comment]
> But GHC 9.4 is already supported in `rhine`. I think to support _9.6_, one needs to bump `dunai`, but I'm not sure. The issue with 9.4 is only that the dependency `monad-bayes` didn't support 9.4 for a long time. But it does now, I think we just need to enable the CI jobs again.

I see, when I tried, I have just updated `flake.lock` and I have got issues I have attributed to GHC 9.4, which in fact where packages set changes.

I have `monad-bayes` on my list, once done, I'll come back to finish the work :)

File: rhine/src/FRP/Rhine/ClSF/Core.hs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {-# LANGUAGE Arrows #-}
 {-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
```
Comment (turion) [inline rhine/src/FRP/Rhine/ClSF/Core.hs:3]: This is not necessary, right?

**Comment by turion** [comment]
Can you fix fourmolu and the remaining build errors?

**Comment by blackheaven** [comment]
I had a fix, I forgot to push it :/

**Comment by blackheaven** [comment]
Is there anything I can do to make the PR move forward?

**Comment by turion** [comment]
Sorry, I overlooked that there was already a PR open to fix this in https://github.com/turion/rhine/pull/249 :( It was open since longer and I wasn't sure it would be merged. Thank you anyways!

**Pull request closed without merging**